### ISSUE TITLE
Deprecate `qiskit.test`

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit#6862]
### What is the expected enhancement?

As a follow up to #6753 we should start to deprecate the `qiskit.test` namespace. Most of the code there is isolated to terra's unittests and is not something we should be exporting as a public interface in the package's namespace. Test code should live as close to the tests as possible the more abstracted away it is the harder it is to debug test failures or understand what a test is actually doing. As some downstream projects use the base test classes they should probably copy them locally because every project's testing needs are different.

We should investigate if there are any things in there we still want to keep and come up with a strategy for preserving them. The only example I can think of currently is `qiskit.test.mock` (https://github.com/Qiskit/qiskit-terra/tree/main/qiskit/test/mock) which has all the fake backends and `ibmq_mock` (https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/test/ibmq_mock.py ) which is used in hidden cells for jupyter executes in docstrings that use the ibmq provider. We can probably move the mock backends to `qiskit.provider` or make a new namespace for them. For `mock_ibmq` we can probably just put it in `utils` or `tools` (not sure what the distinction is there).

**Comment by chriseclectic** [comment]
Many other qiskit packages import the qiskit test cases classes as the base for their own test classes. What should they do instead if this is no longer importable?

**Comment by jakelishman** [comment]
Closed by #11001 for Qiskit 0.46.

*edit*: also, the answer to Chris' question above for others is that `QiskitTestCase` is specific to Qiskit (Terra) - it includes various warning skips and things like that that are appropriate for Terra and not for others - and other libraries should define their own test base if they need something more than the base `unittest.TestCase`.  It's fine to start by vendoring `QiskitTestCase` and cutting out the bits you don't need, though, if that's most convenient.  After #11445, `qiskit.test` will be drastically slimmed down to a state that's more appropriate to pick out the bits to vendor, then another PR like #10998 will actually move `QiskitTestCase` out of the Qiskit package, so it won't be importable.
